p4src/linux_networking/es2k/README_LINUX_NETWORKING.md

```diff
@@ -0,0 +1,240 @@
+
+ p4rt-ctl add-entry br0 linux_networking_control.handle_tx_from_ovs_to_host_table "vmeta.common.vsi=8,hdrs.dot1q_tag[vmeta.common.depth].hdr.vid=1,action=linux_networking_control.remove_vlan_and_send_to_port(1,30)"
+ p4rt-ctl add-entry br0 linux_networking_control.handle_rx_loopback_from_ovs_to_host_table "vmeta.misc_internal.vm_to_vm_or_port_to_port[27:17]=14,user_meta.cmeta.bit32_zeros=0,action=linux_networking_control.set_dest(30)"
+ p4rt-ctl add-entry br0 linux_networking_control.vlan_pop_mod_table "vmeta.common.mod_blob_ptr=1,action=linux_networking_control.vlan_pop"
+
+# Configuration for control packets by removing VLAN ID 2 tagging and sending from VSI-8 to overlay VF with VSI-15
+
+ p4rt-ctl add-entry br0 linux_networking_control.handle_tx_from_ovs_to_host_table "vmeta.common.vsi=8,hdrs.dot1q_tag[vmeta.common.depth].hdr.vid=2,action=linux_networking_control.remove_vlan_and_send_to_port(2,31)"
+ p4rt-ctl add-entry br0 linux_networking_control.handle_rx_loopback_from_ovs_to_host_table "vmeta.misc_internal.vm_to_vm_or_port_to_port[27:17]=15,user_meta.cmeta.bit32_zeros=0,action=linux_networking_control.set_dest(31)"
+ p4rt-ctl add-entry br0 linux_networking_control.vlan_pop_mod_table "vmeta.common.mod_blob_ptr=2,action=linux_networking_control.vlan_pop"
+```
+
+#### 8. Configure rules for underlay control packets.
+Configure rules to send underlay control pakcets from IDPF netdev to physical port.
+
+Below configurations assumes
+- Underlay IDPF netdev has a VSI value 10
+- First physical port will have a port ID of 0
+
+```
+# Configuration for control packets between physical port 0 to underlay IDPF netdev VSI-10
+ p4rt-ctl add-entry br0 linux_networking_control.handle_rx_from_wire_to_ovs_table "vmeta.common.port_id=0,user_meta.cmeta.bit32_zeros=0,action=linux_networking_control.set_dest(26)"
+
+# Configuration for control packets between underlay IDPF netdev VSI-10 to physical port 0
+ p4rt-ctl add-entry br0 linux_networking_control.handle_tx_from_host_to_ovs_and_ovs_to_wire_table "vmeta.common.vsi=10,user_meta.cmeta.bit32_zeros=0,action=linux_networking_control.set_dest(0)"
+```
+
+#### 10. Underlay configuration
+Configure underlay IP addresses, and add static routes.
+
+Below configurations assumes
+- Underlay IDPF netdev has a VSI value 10
+
+```
+p4rt-ctl add-lut-entry br0 linux_networking_control.ecmp_lpm_root_lut "user_meta.cmeta.bit32_zeros=0/255.255.255.255,priority=1,data:ipv4_table_lpm_root=0"
+
+nmcli device set <IDPF netdev for VSI 10> managed no
+ifconfig <IDPF netdev for VSI 10> 40.1.1.1/24 up
+ip route show
+ip route change 40.1.1.0/24 via 40.1.1.2 dev <IDPF netdev for VSI 10>
+```
+
+#### 11. Test the ping scenarios:
+  - Ping between VM's on the same host
+  - Underlay ping
+  - Overlay ping: Ping between VM's on different hosts
+
+## Limitations <a name="limitations"></a>
+Current SAI enablement for the networking recipe has following limitations:

```

Comment:
these limitations are for linux networking P4 and not networking recipe?

Reply:
Done, rephrased it